File: backend/api/schemas/questao.py
from datetime import datetime
from ninja import Schema
from typing import List
from .alternativa import AlternativaIn, AlternativaOut


class QuestaoIn(Schema):
    enunciado: str
    tempo_para_resposta: int
    alternativas: List[AlternativaIn] | None


class QuestaoOut(Schema):
    id: int
    enunciado: str
    alternativas: List[AlternativaOut]
    tempo_para_resposta: int
    created_at: datetime = None
    updated_at: datetime = None

File: frontend/src/types/models/Questao.ts
import { Alternativa } from "./Alternativa";

export interface Questao {
    id?: number;
    enunciado: string;
    alternativas?: Alternativa[];
    tempo_para_resposta?: number;
    created_at?: Date;
    updated_at?: Date;
}

export interface QuestaoPayload {
    id?: number;
    enunciado: string;
    tempo_para_resposta?: number;
    alternativas?: Alternativa[];
}
